owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsVideoUploadsViewModelTest.kt

```diff
@@ -0,0 +1,286 @@
+        }
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select video uploads path - ko - data intent is null`() {
+        videosViewModel.handleSelectVideoUploadsPath(null)
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select video uploads path - ko - folder to upload is null`() {
+        val data: Intent = mockk()
+
+        every { data.getParcelableExtra<OCFile>(any()) } returns null
+
+        videosViewModel.handleSelectVideoUploadsPath(data)
+
+        verify(exactly = 1) {
+            data.getParcelableExtra<OCFile>(UploadPathActivity.EXTRA_FOLDER)
+        }
+    }
+
+    @Test
+    fun `handle select video uploads source path - ok - source path hasn't changed`() {
+        val data: Intent = mockk()
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        // It has to be "" for the test to pass
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+        every { data.getStringExtra(any()) } returns exampleSourcePath
+
+        videosViewModel.loadVideoUploadsSourcePath()
+        videosViewModel.handleSelectVideoUploadsSourcePath(data)
+
+        verify(exactly = 2) {
+            data.getStringExtra(LocalFolderPickerActivity.EXTRA_PATH)
+        }
+        verify(exactly = 1) {
+            preferencesProvider.putString(PreferenceManager.PREF__CAMERA_VIDEO_UPLOADS_SOURCE, exampleSourcePath)
+        }
+    }
+
+    @Test
+    fun `handle select video uploads source path - ok - source path has changed`() {
+        val data: Intent = mockk()
+        val sourcePath = "/New/Source/Path"
+        mockkStatic(PreferenceManager.CameraUploadsConfiguration::class)
+
+        every { preferencesProvider.getString(any(), any()) } returns exampleSourcePath
+        // It has to be "" for the test to pass
+        every { PreferenceManager.CameraUploadsConfiguration.getDefaultSourcePath() } returns ""
+        every { data.getStringExtra(any()) } returns sourcePath
+
+        videosViewModel.loadVideoUploadsSourcePath()
+        videosViewModel.handleSelectVideoUploadsSourcePath(data)
+        val newSourcePath = videosViewModel.getVideoUploadsSourcePath()
+        assertEquals(sourcePath, newSourcePath)
+
+        verify(exactly = 2) {
+            data.getStringExtra(LocalFolderPickerActivity.EXTRA_PATH)
+        }
+        verify(exactly = 1) {
+            cameraUploadsHandlerProvider.updateVideosLastSync(any())
+            preferencesProvider.putString(PreferenceManager.PREF__CAMERA_VIDEO_UPLOADS_SOURCE, sourcePath)
+        }
+    }
+
+    @Test(expected = NullPointerException::class)
+    fun `handle select video uploads source path - ko - data intent is null`() {
+        videosViewModel.handleSelectVideoUploadsSourcePath(null)
+    }
```

Comment:
Does this mean that if it receives a null, the app crashes? We are not handling that 🙉  We should fix it because we should not expect a `NullPointerException` here.